cdap-security-spi/src/main/java/co/cask/cdap/security/spi/authorization/AbstractAuthorizer.java

```diff
@@ -62,16 +61,16 @@ public void enforce(EntityId entity, Principal principal, Action action) throws
   }
 
   @Override
-  public Predicate<EntityId> createFilter(Principal principal) throws Exception {
-    Set<Privilege> privileges = listPrivileges(principal);
-    final Set<EntityId> allowedEntities = new HashSet<>();
-    for (Privilege privilege : privileges) {
-      allowedEntities.add(privilege.getEntity());
-    }
+  public Predicate<EntityId> createFilter(final Principal principal) throws Exception {
     return new Predicate<EntityId>() {
       @Override
       public boolean apply(EntityId entityId) {
-        return allowedEntities.contains(entityId);
+        try {
+          enforce(entityId, principal, new HashSet<>(Arrays.asList(Action.values())));
+          return true;
+        } catch (Exception e) {
```

Comment:
Shouldn't you only return `false` if there's a `UnauthorizedException`?
If there's other types of Exceptions, don't we want to surface those, instead of being silent about the exception?